spinalcordtoolbox/scripts/sct_analyze_texture.py

```diff
@@ -200,8 +200,14 @@ def mean_angle(self):
 
             # Average across angles and save it as wrk_folder/fnameIn_feature_distance_mean.extension
             fname_out = im_m + str(self.param_glcm.distance) + '_mean' + extension
-            run_proc('sct_image -i ' + ' '.join(im2mean_lst) + ' -concat t -o ' + fname_out)
-            run_proc('sct_maths -i ' + fname_out + ' -mean t -o ' + fname_out)
+
+            dim_idx = 3 # img is [x, y, z, angle] so specify 4th dimension (angle)
+
+            img = concat_data(im2mean_lst, dim_idx).save(fname_out)
+            img = Image(fname_out)
+            img.data = np.mean(img.data, dim_idx)
+            img.save()
```

Comment:
`sct_maths -mean` has a check that adds a new axis:

https://github.com/neuropoly/spinalcordtoolbox/blob/073bac3b0307fb90ebc49c98c09058044fa20bea/spinalcordtoolbox/scripts/sct_maths.py#L322-L326

In  case `sct_analyze_texture` is used with 3D data, would it safer to repeat the check before the `np.mean` step here?

Reply:
Small nudge about this @Drulex, and then I can approve. :slightly_smiling_face:

Reply:
Yes good observation